### ISSUE TITLE
Wrong guess_mix when guess=sad

### ISSUE THREAD

**Issue by ymshi449** [psi4/psi4#2410]
There seems to be a wrong HOMO/LUMO mixing when SAD is used. 
```python
H2 =  psi4.geometry(f"""
0 1
H
H 1 3.0
symmetry c1
""")
psi4.set_options({"REFERENCE": "uhf", 
                  "guess": "gwh",
                  "e_convergence": 12,
                 "basis": "cc-pvdz",
                 "guess_mix ": True,
})
E = psi4.energy("scf", molecule=H2)
```
It gives:
```
Mixing alpha HOMO/LUMO orbitals (1,2)
@DF-UHF iter   8:    -0.99872135104496   -1.08802e-14   1.64991e-11 DIIS
```
But for SAD
```python
psi4.set_options({"REFERENCE": "uhf", 
                  "guess": "SAD",
                  "e_convergence": 12,
                 "basis": "cc-pvdz",
                 "guess_mix ": True,
})
E = psi4.energy("scf", molecule=H2)
```
It gives
```
Mixing alpha HOMO/LUMO orbitals (2,3)
@DF-UHF iter   5:    -0.82648407827446   -6.66134e-15   5.36516e-14 DIIS
```
My psi4.__version__ is 1.5.

**Comment by susilehtola** [comment]
The problem is that there are no orbitals in SAD. I've included a tentative fix in #2411.

**Comment by susilehtola** [comment]
As a workaround you can use the `sadno` guess.

**Comment by ymshi449** [comment]
Thank you! Please also fix the documentation.

**Comment by susilehtola** [comment]
> Thank you! Please also fix the documentation.

wym?

**Comment by ymshi449** [comment]
It would also be helpful to specify this in the Psi4 Website.

**Comment by JonathonMisiewicz** [comment]
> It would also be helpful to specify this in the Psi4 Website.

What "this" do you want us to specify? Is "this" that the bug exists? Is "this" that the bug will be fixed in 1.5? Is "this" the workaround with `sadno`?

**Comment by ymshi449** [comment]
To me, it will be helpful to read and know that guess mix does not work with SAD. But since it is planned to be fixed by mixing at the second iteration soon, there is no need to add or modify any documentation then.

**Comment by ymshi449** [comment]
Also, it seems that for `sadno`, `guess_mix` is not implemented.

**Comment by susilehtola** [comment]
> Also, it seems that for `sadno`, `guess_mix` is not implemented.

wym? It should work out-of-the-box, since the `sadno` guess has orbitals like GWH. `huckel` is another possible choice. It might be that DIIS just pulls you back into the spin-restricted solution...

**Comment by JonathonMisiewicz** [comment]
@ymshi449, when you post error reports, _please_ be explicit. We need to know exactly what you are asking and exactly what the problem is. Including the details is more important than the grammar.

> > Also, it seems that for `sadno`, `guess_mix` is not implemented.
> 
> wym? It should work out-of-the-box, since the `sadno` guess has orbitals like GWH. `huckel` is another possible choice. It might be that DIIS just pulls you back into the spin-restricted solution...

@susilehtola 
Using your H2 example

```
  ==> Pre-Iterations <== 

  SCF Guess: Superposition of Atomic Densities' Natural Orbitals via on-the-fly atomic UHF (doi:10.1021/acs.jctc.8b01089).

   -------------------------------------------------------
    Irrep   Nso     Nmo     Nalpha   Nbeta   Ndocc  Nsocc
   -------------------------------------------------------
     A         10      10       1       1       1       0   
   -------------------------------------------------------
    Total      10      10       1       1       1       0   
   -------------------------------------------------------

  ==> Iterations <== 

                           Total Energy        Delta E     RMS |[F,P]|

   @DF-UHF iter   1:    -0.82625214180248   -8.26252e-01   1.90467e-03 DIIS
   @DF-UHF iter   2:    -0.82648067782379   -2.28536e-04   2.30464e-04 DIIS
   @DF-UHF iter   3:    -0.82648407824246   -3.40042e-06   4.94088e-07 DIIS
   @DF-UHF iter   4:    -0.82648407827432   -3.18513e-11   3.29102e-08 DIIS
   @DF-UHF iter   5:    -0.82648407827446   -1.41887e-13   2.57430e-14 DIIS
  Energy and wave function converged.
```

Even though `set guess_mix True` is set in my input file, the output that _should_ appear (and does for `guess gwh` instead of `guess sadno`) when `guess_mix` runs does not.

**Comment by ymshi449** [comment]
Thanks for your advice @JonathonMisiewicz . I will give more explicit questions and comments and more details at the beginning.

**Comment by susilehtola** [comment]
Oh, I see the problem: the mixing happens in `form_C` but SADNO doesn't call that routine in the zeroth iteration. I'm working on a guess agnostic fix.

**Comment by susilehtola** [comment]
This is again good proof for a thorough test suite: the only guess that was tested was GWH...

**Comment by susilehtola** [comment]
The fix is in #2411 for all guess types.
